sled-agent/src/services.rs

```diff
@@ -3929,42 +3994,42 @@ impl ServiceManager {
                     );
                 }
 
-                if let Some(info) = self.inner.sled_info.get() {
-                    zone.add_default_route(info.underlay_address).map_err(
-                        |err| Error::ZoneCommand {
-                            intent: "Adding Route".to_string(),
-                            err,
-                        },
-                    )?;
-                }
-
                 for service in &request.services {
                     let smfh = SmfHelper::new(&zone, service);
 
                     match service {
                         SwitchService::ManagementGatewayService => {
+                            info!(self.inner.log, "configuring MGS service");
                             // Remove any existing `config/address` values
                             // without deleting the property itself.
-                            smfh.delpropvalue("config/address", "*")?;
+                            smfh.delpropvalue_default_instance(
+                                "config/address",
+                                "*",
+                            )?;
 
                             // Restore the localhost address that we always add
                             // when setting up MGS.
-                            smfh.addpropvalue(
+                            smfh.addpropvalue_type_default_instance(
                                 "config/address",
                                 &format!("[::1]:{MGS_PORT}"),
+                                "astring",
```

Comment:
Setting the type always in case a property type hasn't been set by the PropertyGroupBuilder in the first run. The command would return an error otherwise, and there is no harm done if the property type already exists